.github/workflows/checks.yaml

```diff
@@ -86,3 +96,27 @@ jobs:
         run: |
           echo "url=${{ needs.check.outputs.storybook_url }}">> $GITHUB_OUTPUT
           echo "Preview URL: ${{ needs.check.outputs.storybook_url}}"
+
+  analyze-bundle:
+    needs: check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.base_ref }}
+
+      - name: Prepare
+        uses: ./.github/actions/prepare
+
+      - name: Build Target Branch
```

Comment:
we should download artifact for target branch as well, if it doesn't exists, this action should be skipped.